docs/src/content/docs/en/spikes/tech-spikes-support-multiple-instances-for-backend-service.mdx

```diff
@@ -0,0 +1,242 @@
+---
+title: Split multiple instances support
+description: Split multiple instances support
+---
+
+## Background
+
+Currently, heartbeat backend does not support multiple instance which make it hard to scale up the instance when load increasing.
+
+## Expect
+
+Deployment can be achieved through two methods: Docker Compose and k8s.
+
+## Solutions
+
+By introducing volume in Docker Compose and PVC in k8s to achieve data sharing.
+
+### 1. Docker Compose deployment
+
+- docker-compose.yaml
+
+```yaml
+version: "3.4"
+
+services:
+  backend:
+    image: ghcr.io/au-heartbeat/heartbeat_backend:latest
+    expose:
+      - "4322"
+    restart: always
+    deploy:
+      replicas: 3
+    volumes:
+      - file_volume:/files
+  frontend:
+    image: ghcr.io/au-heartbeat/heartbeat_frontend:latest
+    container_name: frontend
+    ports:
+      - "4321:80"
+    depends_on:
+      - backend
+    restart: always
+
+volumes:
+  file_volume:
+```
+
+Three backend containers are started here and the files are mounted to /files
+
+### 2. Kubernetes deployment
+
+#### 2.1 First step
+
+implement init.sh
+
+```shell
+#!/bin/bash
+kubectl delete namespaces heartbeat
+kubectl delete pv pv
+kubectl create namespace heartbeat
```

Comment:
`delete` is a dangerous operation for ops, please assume the user has the related resources